nvflare/private/fed/app/client/worker_process.py

```diff
@@ -32,23 +34,37 @@
 from nvflare.private.fed.client.command_agent import CommandAgent
 
 
+def check_parent_alive(parent_pid, stop_event: threading.Event):
+    while True:
+        if stop_event.is_set():
+            break
+        # TODO: this check only valid in UNIX
+        if os.getppid() != parent_pid:
+            # if parent is not alive, kill its worker process
+            os.killpg(os.getpgid(os.getpid()), 9)
```

Comment:
Maybe sys.exit() is good enough and works on Windows too.

Reply:
When using PyTorch Dataset, we need to do this `os.killpg(os.getpgid(os.getpid()), 9)`
Otherwise only the child process will be killed (the process spawn by torch is not killed.